lib/navigation/navigator.cpp

```diff
@@ -0,0 +1,50 @@
+#include "navigator.hpp"
+
+namespace hyped::navigation {
+
+Navigator::Navigator(){
+  //TODO: impement
+}
+
+void Navigator::navigate(){
+  /*
+  TODO: implement
+
+  main navigation control:
+
+  - look at new data
+  - preprocess
+  - check sensor agreement
+  - calculate current trajecotory
+  - update current trajectory
+  */
+}
+
+void Navigator::publishTrajectory(){
+  /*
+  TODO: implement
+
+  idk how ros works but make sure everyone who needs to know trajectory has it
```

Comment:
This would need a callback function passed to it with the type `std::function<void(hyped::navigation::Trajectory)>` and you would call that function with the current trajectory that you wanna send out

Reply:
Yeah that makes sense, could be a function in the node that calls `getTrajectory` and publishes it at `x` frequency

Reply:
Yeah that was just a placeholder for something that may end up happening. Maybe better to delete it for now since how the nav data will be accessed is still TBD and we may never need it?